spec/controllers/notifications_controller_spec.rb

```diff
@@ -50,5 +53,27 @@
         expect(response).to redirect_to(provider_path(provider_code))
       end
     end
+
+    context "user is not subscribed to notifications" do
+      let(:user) { build(:user, :rolled_over, id: 1) }
+
+      it "calls the user update service" do
```

Comment:
Should we have a spec, in the context where a user does have notifications configured, that the user's state is not updated? You could just move the stub on line 24 into this context and not add a spec, but it might be nice to be explicit about it too.

Reply:
Yep good point. I've added